compiler/resolution/functionResolution.cpp

```diff
@@ -5275,7 +5275,11 @@ static Type* moveDetermineRhsType(CallExpr* call) {
   if (retval == dtVoid) {
     if (CallExpr* rhsCall = toCallExpr(rhs)) {
       if (FnSymbol* rhsFn = rhsCall->resolvedFunction()) {
-        if (rhsFn->hasFlag(FLAG_VOID_NO_RETURN_VALUE) == true) {
+        // `expandExternArrayCalls` can add void assignments when
+        // the extern function has an explicit void return type.
```

Comment:
Is there a way to handle it by adjusting expandExternArrayCalls?

Reply:
Not fully.  If the extern function doesn't specify a return type it is handled correctly there. If it does specify a return type then the type is not known at that point.

```
extern proc foo(a: [] real); // this is handled now
extern proc bar(a: [] real): void; // this void is not resolved yet in expandExternArrayCalls
type t = void;
extern proc baz(a: [] real): t; // this void would be very difficult to resolve at that point
```